hank-server/src/main/java/com/liveramp/hank/ring_group_conductor/RingGroupUpdateTransitionFunctionImpl.java

```diff
@@ -233,9 +239,26 @@ private void removeFromReplicasFullyServing(Map<Domain, Map<Integer, Set<Host>>>
     return result;
   }
 
-  private Integer computeNumReplicasFullyServingRelevantData(Map<Domain, Map<Integer, Set<Host>>> domainToPartitionToHostsFullyServing,
-                                                             Set<DomainAndVersion> domainVersions,
-                                                             Host host) throws IOException {
+  enum LiveReplicaStatus {
+    UNDER_REPLICATED,
+    REPLICATED,
+    OVER_REPLICATED;
+
+    public boolean isFullyReplicated() {
+      return this == REPLICATED || this == OVER_REPLICATED;
+    }
+
+    public static LiveReplicaStatus min(LiveReplicaStatus... statuses) {
```

Comment:
I just realized what I suggested is silly since ordinal order is the actual way that `Enum<E>` order is defined:

```
public final int compareTo(E o) {
        Enum<?> other = (Enum<?>)o;
        Enum<E> self = this;
        if (self.getClass() != other.getClass() && // optimization
            self.getDeclaringClass() != other.getDeclaringClass())
            throw new ClassCastException();
        return self.ordinal - other.ordinal;
    }
```